kong/plugins/apifortress/fortress_serializer.lua

```diff
@@ -0,0 +1,99 @@
+-- ©2016 API Fortress Inc.
+
+local _M = {}
+function string.starts(String,Start)
+   return string.sub(String,1,string.len(Start))==Start
+end
+
+function string.ends(String,End)
+   return End=='' or string.sub(String,-string.len(End))==End
+end
+
+function _M.serialize(ngx)
+  local authenticated_entity
+  if ngx.ctx.authenticated_credential ~= nil then
+    authenticated_entity = {
+      id = ngx.ctx.authenticated_credential.id,
+      consumer_id = ngx.ctx.authenticated_credential.consumer_id
+    }
+  end
+
+	local contentType =  nil
+	local request_headers = {}
+	local reqBody = ngx.req.get_body_data()
```

Comment:
Since this is a hot code path, it is good practise to cache global variables at the main level of this module. See the `alf.lua` serializer for example. This is applicable to all `ngx.*` functions used in hot code paths.